### ISSUE TITLE
Itinerary redesign

### ISSUE THREAD

**Issue by torbent** [unicef/etools-issues#216]
- Update from/to fields with lists of locations for that workspace
- Locations needs to have DSA areas connected to them
- Admin-area should be exposed to create connection for each workspace.
- The following “From”-target should be bound to the previous “TO”-target
- DSA rates should be shown on UI.

**Comment by kasfaw** [comment]
@torbent added to Backlog. Next step should be to see wireframes for us to approve front end ?

**Comment by torbent** [comment]
@janpanchalingam1 @kasfaw here is the proposed wireframe.

![itiniary](https://cloud.githubusercontent.com/assets/688227/24661461/b7491aae-1952-11e7-8d22-71ea8494315c.png)

**Comment by janpanchalingam1** [comment]
@torbent is there a specification associated with this wireframe? Are you able to send us the minutes from the meeting?

**Comment by torbent** [comment]
@janpanchalingam1 the top issue description is the minutes for this task. There is no further specification. The bullet points should however be split into sub-tasks when moved on-deck.

**Comment by janpanchalingam1** [comment]
Please hold until defined by business owner.

**Comment by torbent** [comment]
@janpanchalingam1 @anusenan can we work on this, or split this into two tasks? We are eager to proceed with the location dropdowns, which is the backend part of this. This should be a stand alone task.

**Comment by ilcsik** [comment]
@ewheeler @robertavram I made a PR (https://github.com/unicef/etools/pull/633) contains a refactor for ItineraryItem origin and destination fields, to make them foreign key instead of charfield.

It also contains a migration sequence which tries to match existing origin and destination fields and replace them with the relevant Location ids. If the name is not found in the Location table, the field remains empty. The old values are still accessible in origin_old and destination_old fields.

I have ran a migration on the 04.21. staging dump locally and 46k locations were resolved and migrated to FK, but 65k locations were not matched any Location. 

One solution we suggest is that whenever an FK is missing, we display a readonly field just to show what's the original data, and let the user/admin fill the dropdown with the actual location. Once it's filled, the readonly field disappears.

I'm not sure if this solution is feasible or is there something else we can do in the current situation. Just let me know.

**Comment by janpanchalingam1** [comment]
@torbent yes, please work on this. We can keep as one waffle or split into two, which ever is easier.

**Comment by torbent** [comment]
@janpanchalingam1 @anusenan @ewheeler @robertavram this is blocked by a business decision. Please get back to us with decision or move task to backlog.

**Comment by janpanchalingam1** [comment]
@torbent this is going to be put on hold until we finalize the specifications

**Comment by torbent** [comment]
@janpanchalingam1 or @anusenan please move this card accordingly.
